modules/heat_transfer/src/actions/ThermalContactAction.C

```diff
@@ -70,7 +70,11 @@ ThermalContactAction::validParams()
       "warnings", false, "Whether to output warning messages concerning nodes not being found");
   params.addParam<bool>(
       "quadrature", false, "Whether or not to use quadrature point based gap heat transfer");
-
+  params.addRangeCheckedParam<Real>("penalty",
+                                    1.0,
+                                    "penalty>0",
+                                    "The penalty used in the residual and Jacobian calculations "
+                                    "when using the GapPerfectConductance model");
```

Comment:
add an error when NOT using the GapPerfoectConductance model and the user sets this parameter

Reply:
I have a question about that. I can use isParamSetByUser to add this error check, but what should the `paramError` message be? If I say "This parameter should only be set when using `type = GapPerfectConductance`" it seems incorrect because the penalty parameter is still set to its default value regardless but that's what I'll go with here unless you suggest differently.

Reply:
"should only be set by the user"